up_client_mqtt5_python/mqtt5_utransport.py

```diff
@@ -0,0 +1,303 @@
+
+        if UriValidator.is_local(topic.authority):
+            return f"upl/{entity_id}/{version_major}/{resource_id}"
+
+        try:
+            authority_number = socket.inet_ntop(
+                socket.AF_INET, topic.authority.ip)
+        except ValueError:
+            raise ValueError("Must provide an IP address for the authority.")
+
+        if self.cloud_device:
+            topic_type = "c2d" if msg_type == "send" else "d2c"
+        else:
+            topic_type = "d2c" if msg_type == "send" else "c2d"
+
+        return (f"{topic_type}/{authority_number}/"
+                f"{entity_id}/{version_major}/{resource_id}")
+
+    def send(self, umsg: UMessage) -> UStatus:
+        """
+        Transmits UPayload to the topic using the attributes defined in
+        UTransportAttributes.
+        @param umsg: UMessage to be sent to MQTT
+        @return:Returns OKSTATUS if the payload has been successfully
+        sent (ACK'ed), otherwise it returns FAILSTATUS
+        with the appropriate failure.
+        """
+
+        payload_proto: UPayload = umsg.payload
+        uattributes_proto: UAttributes = umsg.attributes
+
+        publish_properties = mqtt.Properties(mqtt.PacketTypes.PUBLISH)
+        publish_properties.UserProperty = []
+        try:
+            publish_properties.UserProperty.append(
+                ("upriority", UPriority.Name(uattributes_proto.priority)))
+        except ValueError:
+            raise ValueError("Priority not supported.")
+        message_type: UMessageType = uattributes_proto.type
+        publish_properties.UserProperty.append(
+            ("type", UMessageType.Name(message_type)))
+        if message_type == UMessageType.UMESSAGE_TYPE_RESPONSE:
+            publish_properties.UserProperty.append(
+                ("reqId", uattributes_proto.reqid))
+        # TODO: Implement Sink UserProperty with the Short Form URI
+        # if uattributes_proto.HasField("sink"):
+        #     micro_uri_serializer = MicroUriSerializer()
+        #     publish_properties.UserProperty.append(
+        #         ("sink", micro_uri_serializer.serialize(
+        #             uattributes_proto.sink)))
+
+        # if payload.format in [0, 1]:
+        #     # 0: UPAYLOAD_FORMAT_UNSPECIFIED,
+        #     # 1: UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY
+        #     payload_any: Any = payload_proto.data
+        #     payload = payload_any.SerializeToString()
+        # elif payload.format == 2:
+        #     # 2: UPAYLOAD_FORMAT_PROTOBUF
+        #     payload = payload_proto.data.SerializeToString()
+        # elif payload.format == 6:
+        #     # 6: UPAYLOAD_FORMAT_RAW
+        #     payload = payload.data
+        # else:
+        #     # 3: UPAYLOAD_FORMAT_JSON 4: UPAYLOAD_FORMAT_SOMEIP
+        #     # 5: UPAYLOAD_FORMAT_SOMEIP_TLV
+        #     # 7: UPAYLOAD_FORMAT_TEXT
+        #     payload = base64.b64encode(payload_proto.data)
+
+        payload: bytes = payload_proto.SerializeToString()
```

Comment:
The spec still needs to determine how we are serializing the protobuf payload to bytes, but I think this is fine for now